cmd/juno/dbcmd.go

```diff
@@ -0,0 +1,202 @@
+	return dbCmd
+}
+
+func DBInfoCmd() *cobra.Command {
+	return &cobra.Command{
+		Use:   "info",
+		Short: "Retrieve database information",
+		Long:  `This subcommand retrieves and displays blockchain information stored in the database.`,
+		RunE:  dbInfo,
+	}
+}
+
+func DBSizeCmd() *cobra.Command {
+	return &cobra.Command{
+		Use:   "size",
+		Short: "Calculate database size information for each data type",
+		Long:  `This subcommand retrieves and displays the storage of each data type stored in the database.`,
+		RunE:  dbSize,
+	}
+}
+
+func dbInfo(cmd *cobra.Command, args []string) error {
+	dbPath, err := cmd.Flags().GetString(dbPathF)
+	if err != nil {
+		return err
+	}
+
+	if _, err = os.Stat(dbPath); os.IsNotExist(err) {
+		fmt.Println("Database path does not exist")
+		return nil
+	}
+
+	database, err := pebble.New(dbPath)
+	if err != nil {
+		return fmt.Errorf("open DB: %w", err)
+	}
+
+	chain := blockchain.New(database, nil)
+	info := DBInfo{}
+
+	// Get the latest block information
+	headBlock, err := chain.Head()
+	if err != nil {
+		return fmt.Errorf("failed to get the latest block information: %v", err)
+	}
+
+	stateUpdate, err := chain.StateUpdateByNumber(headBlock.Number)
+	if err != nil {
+		return fmt.Errorf("failed to get the state update: %v", err)
+	}
+
+	info.Network = getNetwork(headBlock, stateUpdate.StateDiff)
+	info.ChainHeight = headBlock.Number
+	info.LatestBlockHash = headBlock.Hash
+	info.LatestStateRoot = headBlock.GlobalStateRoot
+
+	// Get the latest L1 block information
+	l1Head, err := chain.L1Head()
+	if err == nil {
+		info.L1Height = l1Head.BlockNumber
+		info.L1BlockHash = l1Head.BlockHash
+		info.L1StateRoot = l1Head.StateRoot
+	} else {
+		fmt.Printf("Failed to get the latest L1 block information: %v\n", err)
+	}
+
+	jsonData, err := json.Marshal(info)
+	if err != nil {
+		return fmt.Errorf("marshal JSON: %w", err)
+	}
+
+	fmt.Println(string(jsonData))
+
+	return nil
+}
+
+func dbSize(cmd *cobra.Command, args []string) error {
+	dbPath, err := cmd.Flags().GetString(dbPathF)
+	if err != nil {
+		return err
+	}
+
+	if dbPath == "" {
+		return fmt.Errorf("--%v cannot be empty", dbPathF)
+	}
+
+	if _, err = os.Stat(dbPath); os.IsNotExist(err) {
+		fmt.Println("Database path does not exist")
+		return nil
+	}
+
+	pebbleDB, err := pebble.New(dbPath)
+	if err != nil {
+		return err
+	}
+
+	var (
+		totalSize  utils.DataSize
+		totalCount uint
+
+		withHistorySize    utils.DataSize
+		withoutHistorySize utils.DataSize
+
+		withHistoryCount    uint
+		withoutHistoryCount uint
+
+		items [][]string
+	)
+
+	for _, b := range db.BucketValues() {
+		fmt.Fprintf(cmd.OutOrStdout(), "Calculating size of %s, remaining buckets: %d\n", b, len(db.BucketValues())-int(b)-1)
+		bucketItem, err := pebble.CalculatePrefixSize(cmd.Context(), pebbleDB.(*pebble.DB), []byte{byte(b)})
+		if err != nil {
+			return err
+		}
+		items = append(items, []string{b.String(), bucketItem.Size.String(), fmt.Sprintf("%d", bucketItem.Count)})
```

Comment:
you don't need to create temp. slice i.e.:
```
items = append(items, b.String(), bucketItem.Size.String(), fmt.Sprintf("%d", bucketItem.Count))
```

Reply:
`items` is `[][]string`, so I had to put the list of strings in `[]string{}` before appending it.

It's used in `table.Appendbulk`.

Reply:
I see, makes sense